src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -60,6 +60,19 @@ protected override Task<ConnectionContext> CreateConnection(string target = null
             return _connectionFactory.ConnectAsync(HubEndpoint, TransferFormat.Binary, ConnectionId, target, headers: CustomHeader);
         }
 
+        protected override Task InnerCloseAsync()
+        {
+            foreach (var key in _clientConnectionManager.ClientConnections.Keys)

```

Comment:
Just loop over the connections and access the key. Using Keys or Values will make a copy of the list while enumerator the collection directly doesn't (but it takes a snapshot).

Reply:
That's exactly what I want, I mean to get a copy.

I will remove the connection from the Dict in the loop, and it's usually not a good practice doing this while iterating the original dataset I guess.

Reply:
Sorry, this method is no longer useful.
We will send a message to Service with `lastAckId` for each `clientConnection` in the previous proposal and it's not necessary now.